providers/reframe.go

```diff
@@ -0,0 +1,39 @@
+package providers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-delegated-routing/client"
+	"github.com/ipfs/go-delegated-routing/gen/proto"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+func NewReframeProviderStore(httpClient *http.Client, endpointURL string) (*reframeProvider, error) {
+	q, err := proto.New_DelegatedRouting_Client(endpointURL, proto.DelegatedRouting_Client_WithHTTPClient(httpClient))
+	if err != nil {
+		return nil, err
+	}
+	return &reframeProvider{
+		reframe: client.NewClient(q),
+	}, nil
+}
+
+type reframeProvider struct {
+	reframe *client.Client
+}
+
+func (x *reframeProvider) AddProvider(ctx context.Context, key []byte, prov peer.AddrInfo) error {
+	return fmt.Errorf("reframe does not support adding providers")
+}
+
+func (x *reframeProvider) GetProviders(ctx context.Context, key []byte) ([]peer.AddrInfo, error) {
+	// XXX: should we use cid.Cast or cid.Decode here?
+	c, err := cid.Cast(key)
+	if err != nil {
+		return nil, err
+	}
+	return x.reframe.FindProviders(ctx, c)
+}
```

Comment:
> 	// XXX: should we use cid.Cast or cid.Decode here?

Neither, the key passed to GetProviders is a multihash, so we need to construct a raw CID here. Since this is just a dumb passthrough, I don't think we should use `multihash.Cast()` either, just pass the bytes dumbly through and let the server determine if it supports the digest.

```suggestion
  func (x *reframeProvider) GetProviders(ctx context.Context, key []byte) ([]peer.AddrInfo, error) {
	return x.reframe.FindProviders(ctx, cid.NewCidV1(cid.Raw, multihash.Multihash(key)))
}
```

Reply:
This breaks the current tests though: the tests that where written when StoreTheIndex was submitted. In these tests the key passed to GetProviders is `cid.NewCidV0(ipfsutil.Hash([]byte("testkey")))`.

Reply:
I guess then that is is a question for STI as to whether they can accept a CIDv1?  CIDv0 would work but I don't think we should be adding new things with CIDv0, right?

Reply:
Why not be explicit here?

DHT provider records are generally provider records (although I don't think that's enforced). So let's do:
1. Try interpreting the key as a multihash (including validating it) by doing `multihash.Cast` and erroring if it's not a multihash
2. Create a CIDv1 `cid.NewCidV1(cid.Raw, multihashKey)`
3. Send that CID off through Reframe

Reply:
I figured out why the old STI test wasn't working. The Mock delegated server is indexing CIDs (not multihashes) and reframe was rewriting the test CIDS from V0 to V1/Raw. It was a mock server issue.

Reply:
@guseggert Ok, Gus. All fixed. Ready for another review. Thanks!

Reply:
> Why not be explicit here?

You mean why not validate the multihash? Because then this will not pass through reqs for hash functions that it doesn't understand, which seems unreasonable to me.

Reply:
I thought that `multihash.Cast()` was looking up the hash function code, but it isn't, so I agree that we should use that instead of `multihash.Multihash()`